ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,270 @@
+        let container = UIStackView(frame: .zero)
+        container.alignment = .center
+        container.distribution = .fill
+
+        return container
+    }()
+
+    private lazy var button: UIButton = {
+        let button = UIButton(type: .system)
+        return button
+    }()
+
+    private let badgeView: UILabel = {
+        let badgeView = UILabel(frame: .zero)
+        badgeView.layer.masksToBounds = true
+        badgeView.backgroundColor = Colors.Palette.dangerPrimary.color
+        badgeView.textColor = .white
+        badgeView.textAlignment = .center
+        badgeView.font = UIFont.systemFont(ofSize: Constants.badgeFontSize, weight: .regular)
+        badgeView.isHidden = true
+
+        return badgeView
+    }()
+
+    private func updateLayout() {
+        updateButton()
+        updateBadgeView()
+        invalidateIntrinsicContentSize()
+    }
+
+    private func updateButton() {
+        button.isEnabled = item.isEnabled
+
+        if isLeftItem {
+            let isRTL = effectiveUserInterfaceLayoutDirection == .rightToLeft
+            button.contentEdgeInsets = UIEdgeInsets(top: 0, left: isRTL ? 0 : Constants.leftBarButtonItemLeadingMargin, bottom: 0, right: isRTL ? Constants.leftBarButtonItemLeadingMargin : 0)
+        } else {
+            button.contentEdgeInsets = UIEdgeInsets(top: 0, left: Constants.rightBarButtonItemHorizontalPadding, bottom: 0, right: Constants.rightBarButtonItemHorizontalPadding)
+        }
+
+        button.tag = item.tag
+        button.tintColor = item.tintColor
+        button.titleLabel?.font = item.titleTextAttributes(for: .normal)?[.font] as? UIFont
+
+        var portraitImage = item.image
+        if portraitImage?.renderingMode == .automatic {
+            portraitImage = portraitImage?.withRenderingMode(.alwaysTemplate)
+        }
+        var landscapeImage = item.landscapeImagePhone ?? portraitImage
+        if landscapeImage?.renderingMode == .automatic {
+            landscapeImage = landscapeImage?.withRenderingMode(.alwaysTemplate)
+        }
+
+        button.setImage(traitCollection.verticalSizeClass == .regular ? portraitImage : landscapeImage, for: .normal)
+        button.setTitle(item.title, for: .normal)
+
+        if let action = item.action {
+            button.addTarget(item.target, action: action, for: .touchUpInside)
+        }
+
+        button.showsLargeContentViewer = true
+
+        if let customLargeContentSizeImage = item.largeContentSizeImage {
+            button.largeContentImage = customLargeContentSizeImage
+        }
+
+        if item.title == nil {
+            button.largeContentTitle = item.accessibilityLabel
+        }
+
+        if #available(iOS 13.4, *) {
+            // Workaround check for beta iOS versions missing the Pointer Interactions API
+            if arePointerInteractionAPIsAvailable() {
+                button.isPointerInteractionEnabled = true
+            }
+        }
+    }
+
+    private func updateBadgeView() {
```

Comment:
don't we have already a badgeview for sidetabbar and tabbarview. we should refactor it and use it for all of these 3 places so the UI is consistent across the framework

Reply:
I totally agree with your suggestion, please give me some time and I'll get back to this as soon as possible because I just got assigned some high-priority tasks for the next week.

Reply:
@harrieshin I have a request that can I create another task for refactoring the badge view? There are two reasons why I am asking this:
1. Currently, I am not touching any part of the codebase except creating the new class i.e BarButtonItemView and so it will not affect anything existing and so no extra testing is required.
2. As, I am quite busy with some high-priority tasks for the notification center feature so I am asking that can we merge this PR, and I can create another PR for the refactoring one where one badge view will be used by TabBarItemView class and BarButtonItemView class.

Reply:
I started to look into what we can refactor and then I realized that there is no need for refactoring because I am already using the existing badge view class i.e BadgeView.swift and then I am just setting/updating the frame for badge view which is similar to what we have done in the case of the side tab bar. Also, there will be no impact on the overall SDK size because anyways setting/updating the frame of badge view is just about 50 lines of code.

I believe that we should be doing setting/updating the frame of badge view separately because of the following reasons:
1. Side/Bottom tab bar buttons can also have titles beneath them whereas UI bar buttons in the navbar will never have separate titles under the button so that makes the difference in setting up the frame of badge view.
2. UI bar buttons in the navbar can be on the right or left side of the navbar title which is not the case in the side/bottom tab bar buttons, which also makes the difference in setting up the frame of badge view.
3. If I make a common function in the already available BadgeView.swift class to setup/update the frame of the badge view then it has to be so generic such that it can handle both the above-mentioned scenarios which I would not be a better code than the current one.

I understand that we should be careful as much as possible in terms of bringing in new functionality and adding a lot more public APIs more than we will utilize but in this scenario, if we notice carefully then we can see that we are not adding a lot more public APIs and definitely not much code as compared to if we would try to make a common method for setting/updating the frame of the badge view.

Reply:
does badgeView.swift accomodate the correct roundedRect corners and the red background that you need for the badge? currently you are creating your own unlabeled in the current PR iteration. how the text count is being handled when text is too long is common logic that can be used for both sidebar and tab bar